### PR TITLE
Bunch of stuff

### PR DESCRIPTION
**Pull request by NANI-SORE**
Features/changes:
- added chewie for video controls, autoplay, looping...
- upped versions of cupertino_icons and video_player dependencies (see pubspec for comments on them)
- videos will autoplay when viewed and autostop when scrolled to the next
- when viewing videos, show their dimensions on the bottom
- snackbars when loading next page or when added tag to search/created new tab
- loading progress to thumbnails and viewer (except videos)
- scrollbar on GridView
- scroll GridView to current item in viewer, this will also trigger next page load when you reach last available rows
- show current/total items counter in viewer
- increased maxScale on images to 8

Fixes:
- set tag search results height to 300, otherwise it was overflowing when > 5 results
- couldn't trigger next page load if first page didn't fill the screen (example: too many thumbnails per row)
- force thumbnailURL on gifs too, (sampleURL can be a gif itself, this will tank the performance)
- don't render empty tag list entries
- don't render items in viewer when they aren't in preloadCount range (I'm not sure if this will affect how the app retrieves these images when returning to them later, probably will force reload when out of range for too long)

### CODE REVIEW

**Comment by grravity** [comment]
Let me know if you want me to make an issue, but I wanted to suggest not having the playback time and bar show after you swipe between videos when it was hidden on the previous image.

So basically it's hidden by default.

Also would it maybe make more sense to utilize the top bar for resolution, maybe under the file counter? That way it's not wasting anymore space? Or my suggestion, read below.

I had mentioned this in my suggestions list issue, but my personal suggestion would to have added an area above the tags where it displayed information about the file. I do encourage looking at how Anime boxes does this (open an image, then hit the info button in top right), because it is pretty clean and allows for more information to be displayed about the file.

Basically they put media info, source url, dimensions, etc. Then there's a HR line if I recall, and below are tags (which are also color coded). 

On a slightly related sidenote, I did think this would show better as a hamburger menu that pulls from the right, instead of a floating window like the tags are currently on. However Noob said he prefers the popup which I respect. Maybe an option would be possible to switch between the two however?

Besides that, this was a great PR. Really polished things. I like being able to see the loading percentage especially. Thanks!

**Comment by NANI-SORE** [comment]
Thanks for the feedback. Sorry for late reply.

> Let me know if you want me to make an issue

Yeah, you probably should, especially if you make feature suggestions. Closed PRs are not really a place for that.

> not having the playback time and bar show after you swipe between videos when it was hidden on the previous image

I'm not sure that's possible now, because I used default controls and they don't really give any control (heh) on how and when we show them. I need to write my own controls element to make it work.

> Also would it maybe make more sense to utilize the top bar for resolution

Resolution info was added as an afterthought (debug thing even), I will do something about it in my next commit. Probably hide or move to the info block, if/when we'll have it.

> this would show better as a hamburger menu ... However Noob said he prefers the popup which I respect

I'm not really in the position to make any big redesign changes, I'm just a random dude who saw this repo and wanted to help with stuff.
But I don't think that making two different designs for the same element is a good idea (at least at the current stage), even though I too don't like current popup style.
Maybe I will make it as an option later, idk.


I really like an idea of an open source booru viewer and I want to do a lot of things for this app, but because of uni and work stuff I don't really have a lot of free time to do anything big right now. Maybe in the future, we will see.

**Comment by NO-ob** [comment]
>but because of uni and work stuff I don't really have a lot of free time to do anything big right now
i know the feel kek, im gonna try get my teacher to let me work on lolisnatcher again this semester for my assignment so I should be able to get a decent amount of work done on it

**Comment by grravity** [comment]
Ah your teacher sounds very cultured. :)
